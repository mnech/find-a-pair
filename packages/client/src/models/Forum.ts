export interface ITopic {
  id: number;
  title: string;
  userName: string;
}

export interface IForumState {
  topicsData: ITopic[];
  error: string | null;
  isLoading: boolean;
}

export interface IForumResponse {
  id: number;
  title: string;
  user: User;
}

export interface User {
  userName: string;
}
