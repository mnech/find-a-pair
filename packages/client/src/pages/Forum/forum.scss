
a {
  display: inline-block;

  color: inherit;
  text-decoration: none;
}

.forum-category {
  background-color: hsl(195, 73%, 58%);

  color: hsl(0, 0%, 100%);
  font-weight: 600;
  font-size: 13px;
  text-shadow: 0 1px 1px hsl(195, 73%, 40%);
}

