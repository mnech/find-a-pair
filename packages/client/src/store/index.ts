import { configureStore, PreloadedState } from '@reduxjs/toolkit';
import combineReducers from '../reducers/index';
let initialState;

if (window! == undefined) {
  initialState = window.initialState;
  delete window.initialState;
}

export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: combineReducers,
    preloadedState,
  });
};

export const store = setupStore(initialState);

export type RootState = ReturnType<typeof combineReducers>;
export type AppStore = ReturnType<typeof setupStore>;
