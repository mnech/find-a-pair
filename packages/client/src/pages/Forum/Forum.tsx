import React from 'react';
import './forum.scss';

const forums = {
  // TODO api
  category: {
    title: 'Category title',
    posts: [
      {
        title: 'Title',
        description: 'Description',
        count: '576',
      },
      {
        title: 'Title2',
        description: 'Description2',
        count: '576',
      },
    ],
  },
  category2: {
    title: 'Category2 title',
    posts: [
      {
        title: 'Title',
        description: 'Description',
        count: '576',
      },
      {
        title: 'Title2',
        description: 'Description2',
        count: '576',
      },
    ],
  },
};

const Forum = () => {
  return (
    <div>
      <header className="border-bottom border-info p-4">Forum</header>

      <div className="p-5">
        {Object.values(forums).map(item => {
          return (
            <div key={item.title} className="mt-4">
              <div className="forum-category rounded-top">
                <div className="p-3">{item.title}</div>
              </div>

              <div>
                <div className="forum-head d-flex flex-row bg-dark text-light">
                  <div className="p-3 flex-grow-1">Forums</div>
                  <div className="p-3">Posts</div>
                </div>

                {item.posts.map(item => {
                  return (
                    <div
                      key={item.title}
                      className="p-3 forum-topic d-flex border-bottom bg-light">
                      <div className="d-flex flex-column flex-grow-1">
                        <span className="text-info fw-bold">
                          <a href="#">{item.title}</a>
                        </span>
                        <span className="text-muted">{item.description}</span>
                      </div>

                      <div>
                        <span className="center">{item.count}</span>
                      </div>
                    </div>
                  );
                })}
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default Forum;
