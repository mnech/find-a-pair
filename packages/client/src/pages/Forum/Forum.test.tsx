import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import Forum from './Forum';
import { store } from '../../store';

describe('Forum page', () => {
  test('renders correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Forum />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
