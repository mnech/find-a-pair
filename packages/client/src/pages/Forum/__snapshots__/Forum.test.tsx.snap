// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forum page renders correctly 1`] = `
<div>
  <div>
    <header
      class="border-bottom border-info p-4"
    >
      Forum
    </header>
    <div
      class="p-5"
    >
      <div
        class="mt-4"
      >
        <div
          class="forum-category rounded-top"
        >
          <div
            class="p-3"
          >
            Category title
          </div>
        </div>
        <div>
          <div
            class="forum-head d-flex flex-row bg-dark text-light"
          >
            <div
              class="p-3 flex-grow-1"
            >
              Forums
            </div>
            <div
              class="p-3"
            >
              Posts
            </div>
          </div>
          <div
            class="p-3 forum-topic d-flex border-bottom bg-light"
          >
            <div
              class="d-flex flex-column flex-grow-1"
            >
              <span
                class="text-info fw-bold"
              >
                <a
                  href="#"
                >
                  Title
                </a>
              </span>
              <span
                class="text-muted"
              >
                Description
              </span>
            </div>
            <div>
              <span
                class="center"
              >
                576
              </span>
            </div>
          </div>
          <div
            class="p-3 forum-topic d-flex border-bottom bg-light"
          >
            <div
              class="d-flex flex-column flex-grow-1"
            >
              <span
                class="text-info fw-bold"
              >
                <a
                  href="#"
                >
                  Title2
                </a>
              </span>
              <span
                class="text-muted"
              >
                Description2
              </span>
            </div>
            <div>
              <span
                class="center"
              >
                576
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mt-4"
      >
        <div
          class="forum-category rounded-top"
        >
          <div
            class="p-3"
          >
            Category2 title
          </div>
        </div>
        <div>
          <div
            class="forum-head d-flex flex-row bg-dark text-light"
          >
            <div
              class="p-3 flex-grow-1"
            >
              Forums
            </div>
            <div
              class="p-3"
            >
              Posts
            </div>
          </div>
          <div
            class="p-3 forum-topic d-flex border-bottom bg-light"
          >
            <div
              class="d-flex flex-column flex-grow-1"
            >
              <span
                class="text-info fw-bold"
              >
                <a
                  href="#"
                >
                  Title
                </a>
              </span>
              <span
                class="text-muted"
              >
                Description
              </span>
            </div>
            <div>
              <span
                class="center"
              >
                576
              </span>
            </div>
          </div>
          <div
            class="p-3 forum-topic d-flex border-bottom bg-light"
          >
            <div
              class="d-flex flex-column flex-grow-1"
            >
              <span
                class="text-info fw-bold"
              >
                <a
                  href="#"
                >
                  Title2
                </a>
              </span>
              <span
                class="text-muted"
              >
                Description2
              </span>
            </div>
            <div>
              <span
                class="center"
              >
                576
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
